autogen.sh

```diff
@@ -14,6 +18,24 @@ error() {
 }
 
 
+########################################################################
+## Parse user arguments
+########################################################################
+
+pup_max_nesting=3
+while test ! -z "$1" ; do
+    if test "$1" = "--pup-max-nesting" ; then
+        shift
+        pup_max_nesting=$1
+	shift
+    fi
```

Comment:
It's a dead loop if wrong argument is supplied since `$1` will never get shifted.

Reply:
Good catch.  I'll fix it.